components/screens/full-pages/ledger/ImportLedger.tsx

```diff
@@ -32,28 +34,46 @@ export default function ImportLedger() {
 
       navigate(`/manage-accounts/ledger/${walletId}/import`);
     } catch (error) {
-      toast.error(
-        "Failed to import account from Ledger device, please unlock and open Kaspa app and try again",
-      );
       captureException(error);
       console.error(error);
+
+      navigate("/ledger-connect-for-import-failed");
     }
   };
 
   useEffect(() => {
-    if (!transport) {
-      const currentUrl = window.location.hash.replace("#", "");
-      navigate("/connect-ledger?redirect=" + currentUrl);
+    const currentUrl = window.location.hash.replace("#", "");
+    if (!transport && !isConnecting) {
+      navigate("/ledger-connect-for-import?redirect=" + currentUrl);
```

Comment:
At this point I think we need better way of handling redirects.

We have it for the unlock and the ledger flow now.

Either we have it the state of the react-router, or we provide another context to hold the value of the redirect?

Reply:
I think it is safe we remain current structure as it only happens in `Unlock` and `Ledger connect` flow.